.github/workflows/test.yml

```diff
@@ -13,9 +13,16 @@ on:
 
 jobs:
   integration-tests:
-    name: Integration tests
+    name: Integration (QA) tests
     runs-on: "n1-standard-32-netssd-preempt"
     timeout-minutes: 45
+    strategy:
+      fail-fast: false
+      matrix:
+        # TODO: eventually use the same strategy we used for unit tests but for qa modules
+        project:
```

Comment:
:x: The integration tests currently run all integration tests from all modules. If we restrict it to integration-tests and update-tests, we'll need to add jobs for other modules with integration tests, eg. gateway and backup/s3.

Reply:
Yeah, I thought about that. And I think we might want to differentiate between QA and integration tests.

Integration tests: per-module integration tests, e.g. testing integration of exporter with ES, backup with S3, etc. 

QA tests: tests the complete project, and thus requires complete packaging of the project.

I mention this because, for example, the integration tests in the ES module don't need the Docker image. It's also not the place to write these kind of tests, as testing with the whole container effectively tests the whole application. In the ES module, you want to test only the minimum scope integration, e.g. the exporter + Elasticsearch. I would propose the same with backup - create your manager in JVM, then test against S3. If you want to test the backup process from a holistic pov, then write it as a QA test. Does that make sense?

Reply:
Absolutely :+1: 

Still, they are named "*IT" and only picked up by failsafe and with `-DskipITs=false`. Would you suggest that we don't do that for per-module integration tests?

Reply:
Wouldn't `-DskipITs=false` skip failsafe?

Reply:
No, it's double negation. We run integration tests if we don't skip them.

Reply:
Oh, my bad, ofc :smile:

Reply:
So you could have an integration test job which is like the unit tests, which runs failsafe over the whole code base per module, minus the QA module, which only builds the module you want and executes `verify` as target instead of test. For integration tests you may want to run on the n8 runners since IT tends to be heavier than units.

Then have a separate QA job which runs only the QA sub modules (and which packages the whole distribution beforehand, including the docker image), on beefier nodes, with tons of forks since some tests are alone 3 minutes.